getgauge/impl_loader.py

```diff
@@ -62,6 +62,12 @@ def _import_file(base_dir, file_path):
     rel_path = os.path.normpath(file_path.replace(base_dir + os.path.sep, ''))
     try:
         module_name = os.path.splitext(rel_path.replace(os.path.sep, '.'))[0]
+        # Resolve relative imports
+        if module_name.startswith('.'):
+            for _ in range(file_path.count('..')):
+                base_dir = os.path.dirname(base_dir).replace("/", os.path.sep).replace("\\", os.path.sep)
+            sys.path.append(base_dir)
```

Comment:
Can we do this without modifying the sys path? This is to avoid unintended side effects. For example.

```python
import os
import sys
import importlib
import inspect
from contextlib import contextmanager

@contextmanager
def temporary_sys_path(path):
    original_sys_path = sys.path[:]
    sys.path.append(path)
    try:
        yield
    finally:
        sys.path = original_sys_path

def _import_file(base_dir, file_path):
    rel_path = os.path.normpath(os.path.relpath(file_path, base_dir))
    try:
        module_name = os.path.splitext(rel_path.replace(os.path.sep, '.'))[0]
        # Resolve relative imports
        if module_name.startswith('.'):
            base_dir = os.path.abspath(os.path.join(base_dir, os.path.dirname(file_path)))
            module_name = module_name.lstrip('.')
            with temporary_sys_path(base_dir):
                m = importlib.import_module(module_name)
        else:
            m = importlib.import_module(module_name)
        # Get all classes in the imported module
        classes = inspect.getmembers(m, lambda member: inspect.isclass(member) and member.__module__ == module_name)
        return classes
    except ImportError as e:
        raise ImportError(f"Could not import module {module_name} from {file_path}: {e}")
```